src/test/java/io/gravitee/policy/oauth2/Oauth2PolicyV4EmulationEngineCnfIntegrationTest.java

```diff
@@ -0,0 +1,483 @@
+        }
+
+        @Override
+        protected void configureHttpClient(final HttpClientOptions options) {
+            Oauth2PolicyV4EmulationEngineCnfIntegrationTest.configureHttpClient(options, gatewayPort());
+
+            final PemKeyCertOptions pemKeyCertOptions = new PemKeyCertOptions();
+            pemKeyCertOptions.setCertPath(Oauth2PolicyV3IntegrationTest.class.getResource("/certificate/client1-crt.pem").getPath());
+            pemKeyCertOptions.setKeyPath(Oauth2PolicyV3IntegrationTest.class.getResource("/certificate/client1-key.pem").getPath());
+            options.setPemKeyCertOptions(pemKeyCertOptions);
+        }
+
+        @Override
+        public void configureApi(final Api api) {
+            Plan oauth2Plan = createOauth2Plan(api);
+
+            OAuth2PolicyConfiguration configuration = new OAuth2PolicyConfiguration();
+            configuration.setOauthResource("dummy-oauth2-resource");
+            configuration.getConfirmationMethodValidation().getCertificateBoundThumbprint().setEnabled(true);
+            try {
+                oauth2Plan.setSecurityDefinition(new ObjectMapper().writeValueAsString(configuration));
+            } catch (JsonProcessingException e) {
+                throw new RuntimeException("Failed to set OAuth2 policy configuration", e);
+            }
+
+            api.setPlans(Collections.singletonList(oauth2Plan));
+        }
+
+        @Override
+        public void configureResources(final Map<String, ResourcePlugin> resources) {
+            Oauth2PolicyV4EmulationEngineCnfIntegrationTest.configureResources(resources);
+        }
+
+        @Test
+        void should_access_api_with_valid_certificate(HttpClient client) throws InterruptedException, URISyntaxException, IOException {
+            wiremock.stubFor(get("/team").willReturn(ok("response from backend")));
+
+            // subscription found is valid
+            whenSearchingSubscription(API_ID, CLIENT_ID, PLAN_ID).thenReturn(Optional.of(fakeSubscriptionFromCache(false)));
+
+            client
+                .rxRequest(HttpMethod.GET, "/test")
+                .flatMap(request -> request.putHeader("Authorization", "Bearer " + DummyOAuth2Resource.TOKEN_SUCCESS_WITH_CNF).rxSend())
+                .flatMapPublisher(response -> {
+                    assertThat(response.statusCode()).isEqualTo(200);
+                    return response.toFlowable();
+                })
+                .test()
+                .await()
+                .assertComplete()
+                .assertValue(body -> {
+                    assertThat(body).hasToString("response from backend");
+                    return true;
+                })
+                .assertNoErrors();
+
+            wiremock.verify(1, getRequestedFor(urlPathEqualTo("/team")));
+        }
+
+        @Test
+        void should_return_401_with_valid_certificate_in_header_but_without_cnf_in_token(HttpClient client)
```

Comment:
For this test, the certificate passed through the header is ignored no? (considering the client does MTLS and the policy hasn't enable header extraction)

Reply:
yes it is.